Frameworks/include/CollectionHelpers.h

```diff
@@ -75,10 +81,39 @@ HRESULT WRLToNSCollection<ABI::Windows::Foundation::Collections::IMap<HSTRING, H
     return S_OK;
 }
 
+// Specified for converting IPropertyValue of String to NSString for Toast Action conversion
+// TODO 8218419: Cannot currently consume values in Objective C projection without first converting to IPropertyValue
+template <>
+HRESULT WRLToNSCollection<ABI::Windows::Foundation::Collections::IPropertySet>(ABI::Windows::Foundation::Collections::IPropertySet* map,
+                                                                               NSDictionary* __autoreleasing* pDictionary) {
+    using NSType = Private::CollectionType<ABI::Windows::Foundation::Collections::IPropertySet>::NSMutableEquivalentType;
+    NSType* collection = [[NSType alloc] init];
+    HRESULT hr = WRLHelpers::ForEach(
+        map,
+        [&collection](const Microsoft::WRL::ComPtr<ABI::Windows::Foundation::Collections::IKeyValuePair<HSTRING, IInspectable*>>& pair,
+                      boolean* stop) {
+            Microsoft::WRL::Wrappers::HString key, value;
+            Microsoft::WRL::ComPtr<IInspectable> inspectableValue;
+            Microsoft::WRL::ComPtr<ABI::Windows::Foundation::IPropertyValue> propertyValue;
+            RETURN_IF_FAILED(pair->get_Key(key.GetAddressOf()));
+            RETURN_IF_FAILED(pair->get_Value(inspectableValue.GetAddressOf()));
+            RETURN_IF_FAILED(inspectableValue.As(&propertyValue));
+            RETURN_IF_FAILED(propertyValue->GetString(value.GetAddressOf()));
+
+            [collection setObject:Strings::WideToNSString(value.Get()) forKey:Strings::WideToNSString(key.Get())];
+            return S_OK;
+        });
+    RETURN_IF_FAILED(hr);
```

Comment:
Moving this up may not actually work if the macro is sufficiently silly: The preprocessor is not as advanced as all that.